moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -218,9 +275,9 @@ bool CurrentStateMonitor::waitForCurrentState(const ros::Time t, double wait_tim
   ros::WallDuration elapsed(0, 0);
   ros::WallDuration timeout(wait_time);
 
-  boost::mutex::scoped_lock lock(state_update_lock_);
```

Comment:
I don't like that you moved the mutex lock inside the while loop. This will induce repeated locking and unlocking. When using getCurrentStateTimeHelper, you could keep the lock outside the while loop, couldn't you?

Reply:
No, I couldn't. It is essential to avoid livelock that would happen because of the lock acquisition happening inside the method `getCurrentStateTime`

Reply:
But you introduced the lock-free helper.

Reply:
Indeed.